src/Faithlife.Utility/EnumerableUtility.cs

```diff
@@ -755,7 +755,50 @@ static IEnumerable<T> DoWhereNotNull(IEnumerable<T?> s)
 			this
 #endif
 				IEnumerable<T1> first, IEnumerable<T2> second) =>
-			ZipImpl(first ?? throw new ArgumentNullException(nameof(first)), second ?? throw new ArgumentNullException(nameof(second)), UnbalancedZipStrategy.Throw);
+			ZipImpl(
+				first ?? throw new ArgumentNullException(nameof(first)),
+				second ?? throw new ArgumentNullException(nameof(second)),
+				(firstElement, secondElement) => (firstElement, secondElement),
+				UnbalancedZipStrategy.Throw);
+
+		/// <summary>
+		/// Combines three same sized sequences.
+		/// </summary>
+		/// <param name="first">An IEnumerable whose elements will be returned as ValueTuple.First.</param>
+		/// <param name="second">An IEnumerable whose elements will be returned as ValueTuple.Second.</param>
+		/// <param name="third">An IEnumerable whose elements will be returned as ValueTuple.Third.</param>
+		/// <returns>A sequence of tuples combining the input items. Throws if the sequences don't have the same number of items.</returns>
+		[SuppressMessage("StyleCop.CSharp.ReadabilityRules", "SA1414:Tuple types in signatures should have element names", Justification = "By design.")]
+		public static IEnumerable<(T1, T2, T3)> Zip<T1, T2, T3>(
+#if NETSTANDARD || NETCOREAPP || NET5_0
```

Comment:
I'm not thrilled about adding a new method that's already obsolete in the latest version of .NET.

This could be
```suggestion
#if !NET6_0_OR_GREATER
```

Reply:
It does provide a bridge between .NET 4.8 and .NET 6.0 for code that has to multi-target.